pkg/network/tracer/tracer.go

```diff
@@ -183,7 +184,8 @@ func newTracer(cfg *config.Config) (_ *Tracer, reterr error) {
 	tr.reverseDNS = newReverseDNS(cfg)
 	tr.usmMonitor = newUSMMonitor(cfg, tr.ebpfTracer, tr.bpfTelemetry)
 
-	if cfg.EnableProcessEventMonitoring {
+	tr.verifyPESIsSupported(currKernelVersion)
+	if tr.cfg.EnableProcessEventMonitoring {
```

Comment:
Updating a member of an object and then immediately checking that member is a bit of an anti-pattern.   I don't think we do it elsewhere, and in general I'd assume that `cfg` is immutable throughout the lifetime of tracer. 

This could be more clearly written as:

```
if tr.cfg.EnableProcessEventMonitoring {
    if !canRunProcessStream(kernelVersion) {
         // log, return error or break
    }
    // initialize process event stream
}
```

Reply:
fixed this